### ISSUE TITLE
[4.0.0-beta.16] Tabs not compatible with fullTemplateTypeCheck

### ISSUE THREAD

**Issue by TomDemulierChevret** [ionic-team/ionic#16492]
# Bug Report

**Ionic Info**
Run `ionic info` from a terminal/cmd prompt and paste the output below.

```
Ionic:

   ionic (Ionic CLI)             : 4.3.1
   Ionic Framework               : @ionic/angular 4.0.0-beta.16
   @angular-devkit/build-angular : 0.10.7
   @angular-devkit/schematics    : 7.0.7
   @angular/cli                  : 7.0.7
   @ionic/angular-toolkit        : 1.2.0

Cordova:

   cordova (Cordova CLI) : not installed
   Cordova Platforms     : not available
   Cordova Plugins       : not available

System:

   NodeJS : v8.10.0 (C:\Program Files\nodejs\node.exe)
   npm    : 6.4.1
   OS     : Windows 10
```

**Describe the Bug**
Since beta.16, any build done with ``ionic cordova build android --prod --release`` will fail if we ``fullTemplateTypeCheck`` is enabled in ``tsconfig.json``.

```json
"angularCompilerOptions": {
   "fullTemplateTypeCheck": true
  }
```

The build fails with the following output :
```
ERROR in src\app\pages\tabs\tabs.component.html(3,5): : Directive TabDelegate, Expected 0 arguments, but got 1.
src\app\pages\tabs\tabs.component.html(6,5): : Directive TabDelegate, Expected 0 arguments, but got 1.
src\app\pages\tabs\tabs.component.html(9,5): : Directive TabDelegate, Expected 0 arguments, but got 1.
src\app\pages\tabs\tabs.component.html(12,5): : Directive TabDelegate, Expected 0 arguments, but got 1.
src\app\pages\tabs\tabs.component.html(15,5): : Directive TabDelegate, Expected 0 arguments, but got 1.
```

The issue is present with the ``tabs`` starter pack and therefore is not related to my own code.

**Steps to Reproduce**
1. Create an Ionic 4 new project with the starter pack ``tabs``
2. Enable ``fullTemplateTypeCheck`` in ``tsconfig.json``
3. Execute ``ionic cordova build android --prod --release``

**Expected Behavior**
The release/prod build with ``fullTemplateTypeCheck`` worked in beta.15 and therefore should still work.

**Comment by manucorporat** [comment]
Fixed by https://github.com/ionic-team/ionic/commit/5f32053edbcaa169e2e349aaa59ea49c379b2585#diff-c576578e8633ac937a861acb67175650L34

**Comment by TomDemulierChevret** [comment]
I can confirm it has been fix'd with ``4.0.0-beta.17`` event if it's not in the changelod @manucorporat. ;)
